### PR TITLE
Obsolete archane functionality

### PR DESCRIPTION
**Pull request by oliver-sanders**
It would make sense to remove cruft and cut down the codebase *before* we spin up major work on Cylc 8 to make things simpler:

* [x] Part 1 - Remove obsolete commands and modules
* [x] Part 2 - Remove obsolete Cylc configuration & documentation
* [ ] Part 3 - Remove outdated back-compat code branches - *future work*

As the communication protocols are changing we can now drop *all* back-compat on the network side.

@cylc/core anything controversial, anything missed?

### CODE REVIEW

File: tests/cylc-5to6/00-simple-start-up.t [hunk 1]
```diff
@@ -28,68 +28,5 @@ contains_ok "${TEST_NAME}.stderr" <<'__ERR__'
...
-            method = background
-            retry delays = PT5M # UPGRADE CHANGE: delays as ISO 8601 durations
-[visualization]
-    initial cycle point = 20140101T00
-    final cycle point = 20140102T06
-__OUT__
-cmp_ok "$TEST_NAME.stderr" </dev/null
-#-------------------------------------------------------------------------------
 purge_suite $SUITE_NAME
 exit
```
Comment (matthewrmshin) [inline tests/cylc-5to6/00-simple-start-up.t]: Delete these tests?

**Comment by matthewrmshin** [comment]
I'll add most of the suite run database restart upgrade logic.

File: lib/cylc/command_polling.py [hunk 2]
```diff
@@ -1,99 +0,0 @@
...
-# (at your option) any later version.
-#
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with this program.  If not, see <http://www.gnu.org/licenses/>.
-"""Encapsulates polling activity for CLI commands."""
```
Comment (matthewrmshin) [inline lib/cylc/command_polling.py]: This module is still used. Travis CI failures confirm.

Comment (hjoliver) [reply lib/cylc/command_polling.py]: It's used in `cylc stop` and `cylc suite-state`.  We probably won't need it in the former anymore thanks to duplex comms, but probably do still need it in the latter (DB not server).

**Comment by kinow** [comment]
Great idea @oliver-sanders !!! 🎉

**Comment by matthewrmshin** [comment]
(Just a few failed tests to sort out. You are almost there.)

**Comment by oliver-sanders** [comment]
Honestly surprised we had tests for the `cylc documentation` command.

**Comment by hjoliver** [comment]
There's also a "Cylc 6 Migration" section in the User Guide.

**Comment by matthewrmshin** [comment]
Agreed. Fail is better than warning in this case. Warnings are just ignored and can cause downstream issues. With the time scale of Cylc 8, we should really aim to provide back compat only for modern Cylc 7 suites. I'd like to draw a line at 7.7 - but happy to stretch it to 7.5.

**Comment by oliver-sanders** [comment]
There's some old syntax in `etc/dev-bin`. Are these files used any more?

**Comment by matthewrmshin** [comment]
Test failing on style.

**Comment by hjoliver** [comment]
> 
> 
> There's some old syntax in `etc/dev-bin`. Are these files used any more?

I guess you mean `defn-order-test.py`; that could be removed.

**Pull request merged**